spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,118 @@ The following metrics are produced for each channel declared by the application
 // ==== Custom message types
 
 // Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
+
+== Publishing messages to a stream from imperative code
+Traditionally, the reactive world and imperative world are separated and operate in parallel. Reactive Messaging deals with streams of data in the reactive world, while the imperative world is pretty much point to point and synchronous communication. However, imperative programme sometimes needs to connect to reactive streams so that responses can be emitted to a destination service. 
+Bridging the two worlds is very valuable thing to do, so that one microservice can use technologies from both environments. For an instance, a JAX-RS resource might want to publish messages to a Reactive Messaging channel. This section is about enabling imperative code to publish messages to a Reactive Messaging channel, so that it can be consumed by a consumer.
+The approach is to introduce the concept of _channel_.
+
+=== Eastablish a channel
+
+In order to publish messages on a _channel_, we need to connect a channel first. `@Channel("channel_name")` is used for this purpose.
+`@Channel` is a qualifier to inject a _channel_ to a CDI bean or any class that supports CDI injection. 
+
+=== Transit messages to a channel
```

Comment:
You can't use "Transit" like this. "Transit messages through a channel" might make sense, but it would be better not to use the word Transit here.

I assume we're talking about sending and receiving messages via channels. It might make more sense to break up this section with two headings.

=== Receiving messages from a channel.

=== Sending messages to a channel

Reply:
changed to sending messaging to a channel